watch_advertisements/web_app.js

```diff
@@ -0,0 +1,141 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Espruino devices publish a UART service by default.
+const nordicUARTService = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
+const bodyTemperatureService = '00001809-0000-1000-8000-00805f9b34fb';
+const testCharacteristic = '0b301809-193e-11eb-adc1-0242ac120002';
+
+let gattServer = undefined;
+let device = undefined;
+let btDeviceId = null;
+let intervals = [];
+let previous_timstamp = null;
+const urlParam = 'bt_device_id';
+const btDeviceIdUrlParam = new URLSearchParams(window.location.search).get(urlParam);
+
+function disableButtons(disabled=true) {
+  $('btn_load_code').disabled = disabled;
+  $('btn_watch_advertisements').disabled = disabled;
+  $('btn_watch_advertisements_no_adv').disabled = disabled;
+}
+
+function onGattDisconnected(evt) {
+  logInfo(`Disconnected from GATT on device ${evt.target.name}.`);
+  assertFalse(gattServer.connected, 'Server connected');
+}
+
+/**
+ * Load the device code to the Espruino IDE.
+ */
+function loadEspruinoDeviceCode() {
+  fetch('device_code.js').then(response => response.text()).then(data => {
+    let url = 'http://www.espruino.com/webide?code=' + encodeURIComponent(data);
+    window.open(url, '_window');
+  });
+}
+
+function listenToAdvertisements(event) {
+  if (intervals.length < 10) {
+    var data = event.serviceData.get(bodyTemperatureService).getUint8();
+    logInfo(`Advertisement received with data: ${data}`);
+    if (previous_timstamp != null) {
+      if (data != advertisingValue) {
+        logError(`Unexpected failure: Value ${data} was received instead of ${advertisingValue}.`);
+      }
+      t = event.timeStamp - previous_timstamp;
+      logInfo(`Interval: ${t}`);
+      intervals.push(t);
+    }
+    previous_timstamp = event.timeStamp;
+  } else if (intervals.length == 10) {
+    mean = intervals.reduce((a, b) => a + b) / intervals.length;
+    logInfo(`Intervals mean: ${mean}`);
+    if (mean > 3 * interval) {
+      logError(`Unexpected failure: mean of intervals surpassed the limit.`);
+    }
+    disableButtons(false);
+    intervals.push(event.timeStamp - previous_timstamp);
+  }
+}
+
+async function watchAdvertisements() {
+  logInfo('Starting test');
+  disableButtons();
+
+  try {
+    logInfo(`Requesting Bluetooth device with service ${nordicUARTService}`);
+    device = await navigator.bluetooth.requestDevice({
+      filters: [
+        { services: [nordicUARTService] }
+      ],
+      optionalServices: [bodyTemperatureService]
+    });
+    logInfo(`Requested Bluetooh device with id: ${device.id} and name: ${device.name}`);
+    btDeviceId = device.id;
+
+    device.addEventListener('gattserverdisconnected', onGattDisconnected);
+  } catch (error) {
+    logError(`Unexpected failure: ${error}`);
+  }
+
+  disableButtons();
```

Comment:
The buttons are already disabled at this point.

Reply:
Removed.